test/index.test.ts

```diff
@@ -20,121 +21,127 @@ describe('gcpMetadata', () => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
+  it('should access a specific metadata property', async () => {
     nock(HOST).get(`${PATH}/${TYPE}/${PROPERTY}`).reply(200, {});
-    getMetadata(PROPERTY, (err, res) => {
-      assert.equal(res!.config.url, `${BASE_URL}/${TYPE}/${PROPERTY}`);
-      assert.equal(res!.config.headers['Metadata-Flavor'], 'Google');
-      done();
-    });
+    const res = await gcp.instance(PROPERTY);
+    assert.equal(res.config.url, `${BASE_URL}/${TYPE}/${PROPERTY}`);
+    assert.equal(res.config.headers['Metadata-Flavor'], 'Google');
   });
 
-  it('should accept an object with property and query fields', (done) => {
+  it('should accept an object with property and query fields', async () => {
     const QUERY = {key: 'value'};
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
     nock(HOST).get(`${PATH}/${TYPE}/${PROPERTY}`).query(QUERY).reply(200, {});
-    getMetadata({property: PROPERTY, params: QUERY}, (err, res) => {
-      assert.equal(JSON.stringify(res!.config.params), JSON.stringify(QUERY));
-      assert.equal(res!.config.headers['Metadata-Flavor'], 'Google');
-      assert.equal(res!.config.url, `${BASE_URL}/${TYPE}/${PROPERTY}`);
-      done();
-    });
+    const res = await gcp.instance({property: PROPERTY, params: QUERY});
+    assert.equal(JSON.stringify(res.config.params), JSON.stringify(QUERY));
+    assert.equal(res.config.headers['Metadata-Flavor'], 'Google');
+    assert.equal(res.config.url, `${BASE_URL}/${TYPE}/${PROPERTY}`);
   });
 
-  it('should extend the request options', (done) => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
+  it('should extend the request options', async () => {
     const options = {property: PROPERTY, headers: {'Custom-Header': 'Custom'}};
     const originalOptions = extend(true, {}, options);
     nock(HOST).get(`${PATH}/${TYPE}/${PROPERTY}`).reply(200, {});
-    getMetadata(options, (err, res) => {
-      assert.equal(res!.config.url, `${BASE_URL}/${TYPE}/${PROPERTY}`);
-      // assert.equal(res!.config.headers['Metadata-Flavor'], 'Google');
-      assert.equal(res!.config.headers['Custom-Header'], 'Custom');
-      assert.ifError(err);
-      assert.deepEqual(options, originalOptions);  // wasn't modified
-      done();
-    });
+    const res = await gcp.instance(options);
+    assert.equal(res.config.url, `${BASE_URL}/${TYPE}/${PROPERTY}`);
+    assert.equal(res.config.headers['Custom-Header'], 'Custom');
+    assert.deepEqual(options, originalOptions);  // wasn't modified
   });
 
-  it('should return the request error', (done) => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
-    nock(HOST).get(`${PATH}/${TYPE}`).reply(500, {});
-    getMetadata(err => {
-      assert.strictEqual(err!.message, 'Unsuccessful response status code');
-      done();
-    });
+  it('should return the request error', async () => {
```

Comment:
nit: here and elsewhere in the tsts; `throw` instead of `return`.

Reply:
fixed